radio/analyzer_engine/task.py

```diff
@@ -31,4 +43,14 @@ def process(self, task, state, limits):
         task_resources.session.downlink.duration = command_data.get_downlink_duration(state.current_downlink_bitrate())
         task_resources.session.power_budget.energy = command_data.get_downlink_energy_consumption(task_resources.session.downlink.duration)
 
-        return TaskData(command_data.telecommand_name(), command_data.is_scheduled(), task_resources, notes)
+        telecommand_name = command_data.telecommand_name()
+
+        if self.is_sleep(task):

```

Comment:
This is wrong on so many levels I do not even know where to begin.

Reply:
Jak to rozwiązać inaczej, bez grubego refactoringu?

Reply:
Fixed :white_check_mark: